sdk/storage/Azure.Storage.DataMovement/src/SingleItemStorageResourceContainer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Common;
+
+namespace Azure.Storage.DataMovement
+{
+    /// <summary>
+    /// An internal container to wrap a single item.
+    /// </summary>
+    internal class SingleItemStorageResourceContainer : StorageResourceContainer
+    {
+        public override Uri Uri => _resourceItem.Uri;
+
+        public override string ProviderId => _resourceItem.ProviderId;
+
+        private StorageResourceItem _resourceItem;
+
+        public SingleItemStorageResourceContainer(StorageResourceItem resourceItem)
+        {
+            Argument.AssertNotNull(resourceItem, nameof(resourceItem));
+            _resourceItem = resourceItem;
+        }
+
+        protected internal override Task CreateIfNotExistsAsync(CancellationToken cancellationToken = default) => Task.CompletedTask;
+
+        protected internal override StorageResourceContainer GetChildStorageResourceContainer(string path)
+        {
+            // This should never be called for a single item
+            throw new InvalidOperationException("Single item container has no children.");
```

Comment:
Add to DataMovement.Errors.cs

Reply:
Done, also cleaned up all the unused errors in `DataMovement.Errors` while I was there.